pkg/scheduler/framework/types.go

```diff
@@ -720,10 +722,23 @@ func max(a, b int64) int64 {
 
 // resourceRequest = max(sum(podSpec.Containers), podSpec.InitContainers) + overHead
 func calculateResource(pod *v1.Pod) (res Resource, non0CPU int64, non0Mem int64) {
+	inPlacePodVerticalScalingEnabled := utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling)
 	resPtr := &res
 	for _, c := range pod.Spec.Containers {
-		resPtr.Add(c.Resources.Requests)
-		non0CPUReq, non0MemReq := schedutil.GetNonzeroRequests(&c.Resources.Requests)
+		var non0CPUReq, non0MemReq int64
+		req := c.Resources.Requests
+		if inPlacePodVerticalScalingEnabled {
```

Comment:
can you add a comment clarifying that resizing initContainers is not allowed, therefore we don't check their resize status.

Reply:
fixed.